cpp/unreal_plugins/UrdfRobot/Source/UrdfRobot/UrdfJointComponent.cpp

```diff
@@ -0,0 +1,487 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "UrdfRobot/UrdfJointComponent.h"
+
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <PhysicalMaterials/PhysicalMaterial.h>
+#include <PhysicsEngine/PhysicsConstraintComponent.h>
+#include <Math/Matrix.h>
+#include <Math/RotationMatrix.h>
+#include <Math/Rotator.h>
+#include <Math/Vector.h>
+#include <UObject/Object.h> // CreateDefaultSubobject
+
+#include "CoreUtils/ArrayDesc.h"
+#include "CoreUtils/Assert.h"
+#include "CoreUtils/InputActionComponent.h"
+#include "CoreUtils/Log.h"
+#include "CoreUtils/Unreal.h"
+#include "UrdfRobot/UrdfLinkComponent.h"
+#include "UrdfRobot/UrdfParser.h"
+
+// useful for debugging fetch.urdf
+const std::map<std::string, std::pair<std::string, std::vector<double>>> INPUT_ACTIONS = {
+    {"One",        {"add_torque_in_radians",             { 1000.0,  0.0,  0.0}}},
+    {"Two",        {"add_torque_in_radians",             {-1000.0,  0.0,  0.0}}},
+    {"Three",      {"add_force",                         {   50.0,  0.0,  0.0}}},
+    {"Four",       {"add_force",                         {  -50.0,  0.0,  0.0}}},
+    {"Five",       {"add_to_angular_velocity_target",    {    0.1,  0.0,  0.0}}},
+    {"Six",        {"add_to_angular_velocity_target",    {   -0.1,  0.0,  0.0}}},
+    {"Seven",      {"add_to_linear_velocity_target",     {    0.1,  0.0,  0.0}}},
+    {"Eight",      {"add_to_linear_velocity_target",     {   -0.1,  0.0,  0.0}}},
+    {"Nine",       {"add_to_angular_orientation_target", {    0.0,  0.0,  2.0}}},
+    {"Zero",       {"add_to_angular_orientation_target", {    0.0,  0.0, -2.0}}},
+    {"Underscore", {"add_to_linear_position_target",     {    2.0,  0.0,  0.0}}},
+    {"Equals",     {"add_to_linear_position_target",     {   -2.0,  0.0,  0.0}}}
+};
+
+UUrdfJointComponent::UUrdfJointComponent()
+{
+    SP_LOG_CURRENT_FUNCTION();
+
+    // UInputActionComponent
+    input_action_component_ = CreateDefaultSubobject<UInputActionComponent>(Unreal::toFName("input_action_component"));
+    SP_ASSERT(input_action_component_);
+    // Need to explicitly set this up so that the component hierarchy is well-defined.
+    input_action_component_->SetupAttachment(this);
```

Comment:
Do we need to call `RegisterComponent` here?

Reply:
When we create components using `CreateDefaultSubobject` (in the constructor), we don't need to call `RegisterComponent` (as far as I know, I haven't used it

Reply:
Got it.